### ISSUE TITLE
Cargo fmt?

### ISSUE THREAD

**Issue by RCasatta** [Kixunil/payjoin#2]
What do you think of applying `cargo fmt` to the repo?

**Comment by Kixunil** [comment]
Not a fan of `cargo fmt` TBH as I saw it often doing horribly-looking code but will check how it looks. Maybe it can be avoided with reasonably tweaked settings.

**Comment by Kixunil** [comment]
Yeah, as I expected, the formatting is not great and the tool has billion of options, which I currently don't want to handle.
It'd be great if it could whitelist all transformations, then the changes could be introduced slowly.

At least inspired me to do c41101dc6e7d1073883176066970d8aaeab44cda I'd also like to break up long combinator chains but don't feel like doing it manually right now.

**Comment by RCasatta** [comment]
I do agree sometimes it's not the best-looking but to me, it's cool to have a deterministic way to do it for many reasons...

**Comment by Kixunil** [comment]
I'm going to write informal and likely incomplete list of requirements for `cargo fmt` settings and if someone can come up with a configuration satisfying them then I'm in.

* Line length maybe 160 or something
* Break up long combinator chains, not sure what "long" means, perhaps more than two components || many chars
* Commas in `{ a, b, }` always (both horizontal and vertical)
* Never break match *patterns* into multiple lines
* No sorting
* Not sure if I want match arms to be consistent regarding blocks
* Spaces around all arithmetic operators (except after negation), equal signs (except in bounds like `: Iterator<Item=Foo>`), NOT before `?`

Plus general contribution rule: if function and function-like macro arguments are too long and `fmt` suggests breaking them, try instead find a different way to shorten it. (e.g. assign variables beforehand)
